qdrant-landing/content/blog/qdrant-relari.md

```diff
@@ -0,0 +1,289 @@
+
+```python
+k_values = [3, 5, 7, 9]  # Define the different values of top k to experiment
+
+semantic_retrievers = {}
+semantic_logs = {}
+
+# Run the retrievers on the dataset and log retrieved chunks
+for k in k_values:
+    retriever = db.as_retriever(search_type="similarity", search_kwargs={"k": k})
+    log = log_retriever_results(retriever, dataset)
+
+    semantic_retrievers[f"k_{k}"] = retriever
+    semantic_logs[f"k_{k}"] = log
+    print(f"Results on {dataset.name} by Semantic Retriever with k={k} saved!")
+```
+
+#### Send results to Relari
+
+Submit the results to Relari for evaluation. Relari will run Precision/Recall and Rank-Aware metrics on your results. You can benchmark different results in the UI.
+
+```python
+from relari import Metric
+
+for k in k_values:
+    eval_name = f"Semantic Retriever Evaluation k={k}"
+    eval_data = semantic_logs[f"k_{k}"]
+
+    eval_info = client.evaluations.submit(
+        project_id=proj["id"],
+        dataset=dataset_info["id"],
+        name=eval_name,
+        pipeline=[Metric.PrecisionRecallF1, Metric.RankedRetrievalMetrics],
+        data=eval_data,
+    )
+    print(f"{eval_name} submitted!")
+```
+
+In this example, we can see that with our dataset, if we want the recall to be greater than 85%, we should pick a K value of at least 7.
+
+![Top K](/blog/qdrant-relari/topk.png)
+
+We can even look at individual cases in the UI to get more insight into why our retriever failed.
+
+![Top K UI](/blog/qdrant-relari/topk2.png)
+
+Relari and Qdrant can also be combined to evaluate your hybrid search. With this information, you can further tweak the parameter or try a different RAG architecture to select the best-performing strategies for your dataset.
+
+## Auto Prompt Optimization
+
+In conversational applications like chatbots, **Auto Prompt Optimization (APO)** improves the app’s ability to communicate. This technique adapts the chatbot's responses over time, learning from past interactions to adjust phrasing for better results.
+
+For example, in a customer service chatbot, the phrasing of questions can significantly affect user satisfaction. Consider ordering in French versus English in a Parisian cafe—the same information may be transmitted, but it will be received differently.
+
+![French Cafe](/blog/qdrant-relari/cafe.jpg)
+
+Auto Prompt Optimization continuously refines the chatbot’s responses to improve user interactions. Here's how you can implement APO with Relari:
+
+### Set up base prompt
+
+With Auto Prompt Optimization, you can define a system prompt and inspect the results at every iteration of the interaction.
+
+```python
+from relari.core.types import Prompt, UserPrompt
+
+base_prompt = Prompt(
+    system="You are a GitLab legal policy Q&A bot. Answer the following question given the context.",
+    user=UserPrompt(
+        prompt="Question: $question\n\nContext:\n$ground_truth_context",
```

Comment:
The prompt uses some variables (`$question`, `$ground_truth_context`). I think it would make sense to list them and describe their meaning in a few words or at least link some Relari resources.